### PR TITLE
Telegramchat

### PR DESCRIPTION
**Pull request by richlegrand**

### CODE REVIEW

File: src/kritter/ktextclient.py [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 
-    def text(self, to, text):
-        pass
+class Image:
+    def __init__(self, image):
+        self.image = image
 
-    def image(self, to, image):
-        pass
+class KtextClient:
```
Comment (richlegrand) [inline src/kritter/ktextclient.py:16]: I wanted to change the vizy class TelegramDialog to get generic.  This class is tied to Telegram, so it stays as TelegramClient.  Please change back.

File: src/kritter/ktextvisor.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    def send_responses(self, sender, responses):
+        content = [r.content for r in responses] # extract content
+        content = format_content(content) # combine strs and dicts
+        content = [format_dict(c) if isinstance(c, dict) else c for c in content] # format dicts into strs
+        content = format_content(content) # combine strs again
+        with self.lock:
+            self.text_client.send(content, sender)
+
+    def native_callback(self, words, sender, context):
+        # userids need to be saved as strings to avoid
```
Comment (richlegrand) [inline src/kritter/ktextvisor.py]: userid's are unique, yes?  Why do you need to make them strings?  
I don't think you need to call config.load().  You load it in __init__ (why would it change?)

File: src/kritter/ktextvisor.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+                output = f"{sender['name']} is now subscribed"
+            else:
+                output = f"{sender['name']} is already subscribed"
+            self.config['subscribers'] = subscribers
+            self.config.save()
+            return output
+
+        def unsubscribe(words, sender, context):
+            output = "error unsubscribing"
+            self.config.load()
```
Comment (richlegrand) [inline src/kritter/ktextvisor.py]: I don't think you need to call config.load().  You load it in __init__ (why would it change?)

File: src/kritter/ktextvisor.py [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+            "subscribe": (subscribe, "Subscribe self to Vizy updates."),
+            "unsubscribe": (unsubscribe, "Unsubscribe self from Vizy updates.")})
+        @self.callback_receive()
+        def func(words, sender, context):
+            return tv_table.lookup(words, sender, context)            
+
+    def send(self, msg, to):
+        if to is None:
+            # send to all subscribers using their Ids
+            self.config.load()
```
Comment (richlegrand) [inline src/kritter/ktextvisor.py]: I don't think you need to call config.load().  You load it in __init__ (why would it change?)

File: src/kritter/ktextvisor.py (hunk 2 above)
Comment (peregoywork) [reply src/kritter/ktextvisor.py]: config.load() -- removed. I thought there would be an data mismatch from having the users removed in the texting dialog

unique userIds -- an issue of duplicates from different types. When json.loads() returns, the dict in config['subscribers'] will have casted the integer to a string. Users will be allowed to subscribe twice, once with their id as an integer and once with their id as a string. Ex -- { 10 : 'me', '10' : 'me' }.. an alternative would be casting the config dictionary keys upon load.

**Pull request merged**